### PR TITLE
add sphinx docs for sequences

### PR DESCRIPTION
**Pull request by leosartaj**
Added sphinx docs for sequences introduced in #9435.

### CODE REVIEW

**Comment by leosartaj** [comment]
Ping @jcrist @flacjacket

**Comment by aktech** [comment]
These lines: [L653](https://github.com/sympy/sympy/blob/master/sympy/series/sequences.py#L653) and [L656](https://github.com/sympy/sympy/blob/master/sympy/series/sequences.py#L656) creates unneccessary split in code block (in docs). There are few more such cases, except for the intentional spit blocks.

File: sympy/series/sequences.py [hunk 1]
```diff
@@ -625,8 +625,9 @@ def coeff_mul(self, coeff):
 
 def sequence(seq, limits=None):
     """Returns appropriate sequence object.
-    If seq is a sympy sequence, returns SeqPer object
-    otherwise returns SeqFormula object
+
+    If ``seq`` is a sympy sequence, returns :class:`SeqPer` object
+    otherwise returns :class:`SeqFormula` object
```
Comment (aktech) [inline sympy/series/sequences.py]: Grammar (nit pick): add a Full stop at the end of the line.

File: sympy/series/sequences.py [hunk 2]
```diff
@@ -721,9 +722,9 @@ class SeqAdd(SeqExprOp):
 
     Rules:
         * The interval on which sequence is defined is the intersection
-        of respective intervals of sequences.
-        * Anything + EmptySequence, remains unchanged
-        * Other rules are defined in _add methods of sequence classes
+            of respective intervals of sequences.
+        * Anything + :class:`EmptySequence` remains unchanged
+        * Other rules are defined in ``_add`` methods of sequence classes
```
Comment (aktech) [inline sympy/series/sequences.py]: Same as [above](https://github.com/sympy/sympy/pull/9590/files#r34171747), there are few more such cases not visible in diff.

Comment (leosartaj) [inline sympy/series/sequences.py]: I review them once again. Thanks!

**Comment by leosartaj** [comment]
I have updated the docstrings. @aktech

File: sympy/series/sequences.py [hunk 3]
```diff
@@ -40,9 +40,9 @@ def _start_key(expr):
         return start
 
     def _intersect_interval(self, other):
-        """
-        returns the start, stop
-        takes intersection over the two intervals
+        """returns start and stop.
```
Comment (aktech) [inline sympy/series/sequences.py]: *Returns

File: sympy/series/sequences.py [hunk 4]
```diff
@@ -716,14 +714,13 @@ def length(self):
...
-    """
-    Represents term-wise addition of sequences
+    """Represents term-wise addition of sequences.
 
     Rules:
         * The interval on which sequence is defined is the intersection
-        of respective intervals of sequences.
-        * Anything + EmptySequence, remains unchanged
-        * Other rules are defined in _add methods of sequence classes
+            of respective intervals of sequences.
```
Comment (aktech) [inline sympy/series/sequences.py]: It's visible as following: (extra indented)

leo

File: sympy/series/sequences.py [hunk 5]
```diff
@@ -824,16 +820,16 @@ def _eval_coeff(self, pt):
...
     Handles multiplication of sequences only. For multiplication
-    with other objects see ``SeqBase.coeff_mul``.
+    with other objects see :func:`SeqBase.coeff_mul`.
 
     Rules:
         * The interval on which sequence is defined is the intersection
-        of respective intervals of sequences.
-        * Anything * EmptySequence returns EmptySequence
-        * Other rules are defined in _mul methods of sequence classes
+            of respective intervals of sequences.
```
Comment (aktech) [inline sympy/series/sequences.py]: Same as [this](https://github.com/sympy/sympy/pull/9590/files#r34243975).

**Comment by aktech** [comment]
I think It looks good to me, except for few comments.

**Comment by leosartaj** [comment]
Done. I have squashed the commits. @aktech

**Comment by aktech** [comment]
Will commit in 24hr if there is no objection.

**Comment by aktech** [comment]
Thanks its in!

**Pull request merged**